plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/ServicesAndRoutingPage.java

```diff
@@ -250,11 +258,17 @@ public void handleEvent(){
 			String message = "Edit the port to be exposed by the service";
 			final IServicePort port = model.getSelectedServicePort();
 			final ServicePortAdapter target = new ServicePortAdapter(port);
-			ServicePortDialog dialog = new ServicePortDialog(target, message, model.getServicePorts());
+			final boolean previousRoutePort = model.getRoutingPort() == port;
+			ServicePortDialog dialog = new ServicePortDialog(target, message, model.getServicePorts(), previousRoutePort);
 			if(Window.OK == dialog.open()) {
 				target.setName(NLS.bind("{0}-tcp", target.getPort()));
 				model.updateServicePort(port, target);
 				model.setSelectedServicePort(target);
+				boolean routePort = dialog.isRoutePort();
+				if (routePort != previousRoutePort) {
+				    model.setRoutingPort(routePort?target:null);
+				    portsViewer.refresh();
```

Comment:
No because the table now displays both the ports model and the global flag that represents the port selected to the route (last column of the table).